### PR TITLE
sql,adapter: expose source sizes in mz_sources and SHOW SOURCES

### PR DESCRIPTION
**Pull request by benesch**
Per [feedback from @morsapaes](https://github.com/MaterializeInc/materialize/issues/11231#issuecomment-1257999006).

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Review by morsapaes — APPROVED** [review]
🎉

**Comment by benesch** [comment]
This is technically a breaking change to the load gen syntax, so needs to wait until Weds to merge.

**Review by mjibson — APPROVED** [review]

**Comment by benesch** [comment]
I got confused. This PR was not folded into #15052. It just needed to wait until today's breaking change release was cut. Marking for automerge.

File: test/sqllogictest/source_sizing.slt
Comment (benesch) [reply test/sqllogictest/source_sizing.slt:21]: It sure does! Sorry about that. Thanks for fixing.

**Pull request merged**
